### PR TITLE
Colorize gem updates

### PR DESCRIPTION
**Pull request by bronzdoc**
closes https://github.com/bundler/bundler/issues/4334
alt tag

### CODE REVIEW

**Comment by seanlinsley** [comment]
There's this nice bit of code that already exists to colorize text: https://github.com/bundler/bundler/blob/master/lib/bundler/ui/shell.rb. Shouldn't we reuse that here?

The Bundler output is also theoretically supposed to be computer-parseable, which is why color can be disabled. So if this doesn't [follow the same pattern](https://github.com/bundler/bundler/blob/master/lib/bundler/ui/shell.rb#L12-L14), it might be considered a breaking change.

**Comment by bronzdoc** [comment]
@seanlinsley i don't think i can use those shell methods for this specific task or maybe i could but will be uglier, i may create a helper in the shell ui to follow the pattern, something like:
```ruby
 #.../ui/shell.rb
def add_color(str, color)
  @shell.set_color(str, color)
end 
```    
```ruby
     #lib/bundler/source.rb

     def version_message(spec)
      message = "#{spec.name} #{spec.version}"

      if Bundler.locked_gems
        locked_spec = Bundler.locked_gems.specs.find {|s| s.name == spec.name }
        locked_spec_version = locked_spec.version if locked_spec
        if locked_spec_version && spec.version != locked_spec_version
          message += " (#{Bundler.ui.add_color("was #{locked_spec_version}", :green)})"
        end
      end

      message
    end
```
will update in a bit thanks!

@indirect @segiddins thoughts? :grin:

**Comment by segiddins** [comment]
@bronzdoc yeah that seems good

File: lib/bundler/ui/shell.rb [hunk 1]
```diff
@@ -17,6 +17,10 @@ def initialize(options = {})
         @warning_history = []
       end
 
+      def add_color(string, color)
```
Comment (segiddins) [inline lib/bundler/ui/shell.rb]: also needs to be added to the other UI subclass, no?

Comment (bronzdoc) [inline lib/bundler/ui/shell.rb]: yup to ```ruby UI::Silent``` too, thanks.

**Comment by segiddins** [comment]
👍🏻

**Comment by segiddins** [comment]
@homu r+

**Pull request merged**